generate_models.py

```diff
@@ -488,47 +526,681 @@ def generate_commodities(paths, param):
+
+    # Create a dataframe for the start regions
+    Region_start = gpd.GeoDataFrame(grid_cleaned[['l_id', 'V1']], geometry='V1', crs='').rename(
+        columns={'V1': 'geometry'})
+    Region_start.crs = {'init': 'epsg:4326'}
+    Region_start.crs = regions[['NAME_SHORT', 'geometry']].crs
+    Region_start = gpd.sjoin(Region_start, regions[['NAME_SHORT', 'geometry']], how='left', op='intersects')[
+        ['NAME_SHORT']].rename(columns={'NAME_SHORT': 'Region_start'})
+
+    # Create a dataframe for the end regions
+    Region_end = gpd.GeoDataFrame(grid_cleaned[['l_id', 'V2']], geometry='V2', crs='').rename(
+        columns={'V2': 'geometry'})
+    Region_end.crs = {'init': 'epsg:4326'}
+    # regions is in GRS80 which is almost the same as WGS 84 also known as epsg:4326
+    Region_end.crs = regions[['NAME_SHORT', 'geometry']].crs
+    Region_end = gpd.sjoin(Region_end, regions[['NAME_SHORT', 'geometry']], how='left', op='intersects')[
+        ['NAME_SHORT']].rename(columns={'NAME_SHORT': 'Region_end'})
+
+    # Join dataframes
+    grid_regions = grid_cleaned.copy()
+    grid_regions.drop(['V1', 'V2'], axis=1, inplace=True)
+    grid_regions = grid_regions.join([Region_start, Region_end])
+
+    intra = len(grid_regions.loc[(grid_regions['Region_start'] == grid_regions['Region_end']) & ~(
+        grid_regions['Region_start'].isnull())])
+    extra = len(grid_regions.loc[grid_regions['Region_start'].isnull() | grid_regions['Region_end'].isnull()])
+    inter = len(grid_regions) - intra - extra
+
+    # Show numbers of intraregional, interregional and extraregional lines
+    print("\nLinetypes : ")
+    print((("intraregional", intra), ("interregional", inter), ("extraregional", extra)))
+
+    # Remove intraregional and extraregional lines
+    icls_concatenated = grid_regions.loc[
+                        (grid_regions['Region_start'] != grid_regions['Region_end']) &
+                        ~(grid_regions['Region_start'].isnull() | grid_regions['Region_end'].isnull())
+                        ].copy()
+
+    # Sort alphabetically and reindex
+    icls_reversed = reverse_lines(icls_concatenated)
+    icls_reversed.sort_values(['Region_start', 'Region_end', 'tr_type'], inplace=True)
+    icl = icls_reversed.set_index(['Region_start', 'Region_end', 'tr_type'])
+    icl = icl.reset_index()
+
+    icl_final = deduplicate_lines(icl)
+
+    # Transmission evrys, urbs
+    evrys_transmission, urbs_transmission = format_transmission_model(icl_final, paths, param)
+
+    # Ouput
+    urbs_transmission.to_csv(paths["urbs_transmission"], sep=';', decimal=',')
+    print("File Saved: " + paths["urbs_transmission"])
+
+    evrys_transmission.to_csv(paths["evrys_transmission"], sep=';', decimal=',')
+    print("File Saved: " + paths["evrys_transmission"])
+
+
+    #######################################################
+    #    Create shapefile for urbs transmission line ?    #
+    #######################################################
+
```

Comment:
Should we save the urbs transmission lines as a shapefile?

Reply:
we can do that later